autofit/optimize/non_linear/nested_sampling/dynesty.py

```diff
@@ -0,0 +1,527 @@
+
+    @property
+    def evidence(self):
+        """The Bayesian evidence estimated by the nested sampling algorithm.
+
+        For Dynesty, this is computed from the pickled sample's list of all evidence estimates."""
+        return np.max(self.results.logz)
+
+    @property
+    def most_probable_vector(self):
+        """ The median of the probability density function (PDF) of every parameter marginalized in 1D, returned
+        as a list of values.
+
+        If the chains are sufficiently converged this is estimated by passing the accepted samples to *GetDist*, else
+        a crude estimate using the mean value of all accepted samples is used."""
+        if self.pdf_converged:
+            return self.pdf.getMeans()
+        else:
+            return list(np.mean(self.results.samples, axis=0))
+
+    @property
+    def most_likely_index(self):
+        """The index of the accepted sample with the highest likelihood, e.g. that of best-fit / most_likely model."""
+        return np.argmax(self.results.logl)
+
+    @property
+    def most_likely_vector(self):
+        """ The best-fit model sampled by the non-linear search (corresponding to the maximum log-likelihood), returned
+        as a list of values.
+
+        The vector is read from the pickled sampler instance, by first locating the index corresponding to the highest
+        likelihood accepted sample."""
+        return self.results.samples[self.most_likely_index]
+
+    def vector_at_sigma(self, sigma):
+        """ The value of every parameter marginalized in 1D at an input sigma value of its probability density function
+        (PDF), returned as two lists of values corresponding to the lower and upper values parameter values.
+
+        For example, if sigma is 1.0, the marginalized values of every parameter at 31.7% and 68.2% percentiles of each
+        PDF is returned.
+
+        This does not account for covariance between parameters. For example, if two parameters (x, y) are degenerate
+        whereby x decreases as y gets larger to give the same PDF, this function will still return both at their
+        upper values. Thus, caution is advised when using the function to reperform a model-fits.
+
+        For *Dynesty*, this is estimated using *GetDist* if the chains have converged, by sampling the density
+        function at an input PDF %. If not converged, a crude estimate using the range of values of the current
+        physical live points is used.
+
+        Parameters
+        ----------
+        sigma : float
+            The sigma within which the PDF is used to estimate errors (e.g. sigma = 1.0 uses 0.6826 of the PDF)."""
+        limit = math.erf(0.5 * sigma * math.sqrt(2))
+
+        if self.pdf_converged:
+            densities_1d = list(
+                map(lambda p: self.pdf.get1DDensity(p), self.pdf.getParamNames().names)
+            )
+
+            return list(map(lambda p: p.getLimits(limit), densities_1d))
+        else:
```

Comment:
Don't bother putting else when you return in the if

Reply:
Done